geometry/intermediate.py

```diff
@@ -0,0 +1,60 @@
+    """
+    Checks to see if two line segments have at least one point that they share
+    """
+    line_segment_2_start_orientation_relative_to_line_segment_1 = calculate_point_triplet_orientation(line_segment_1.start, line_segment_1.end, line_segment_2.start)
+    line_segment_2_end_orientation_relative_to_line_segment_1 = calculate_point_triplet_orientation(line_segment_1.start, line_segment_1.end, line_segment_2.end)
+    line_segment_1_start_orientation_relative_to_line_segment_2 = calculate_point_triplet_orientation(line_segment_2.start, line_segment_2.end, line_segment_1.start)
+    line_segment_1_end_orientation_relative_to_line_segment_2 = calculate_point_triplet_orientation(line_segment_2.start, line_segment_2.end, line_segment_1.end)
+
+    does_line_segment_2_have_one_end_clockwise_to_line_segment_1 = line_segment_2_start_orientation_relative_to_line_segment_1 * line_segment_2_end_orientation_relative_to_line_segment_1 < 0
+    does_line_segment_1_have_one_end_clockwise_to_line_segment_2 = line_segment_1_start_orientation_relative_to_line_segment_2 * line_segment_1_end_orientation_relative_to_line_segment_2 < 0
+
+    if does_line_segment_2_have_one_end_clockwise_to_line_segment_1 and does_line_segment_1_have_one_end_clockwise_to_line_segment_2:
+        return True
+
+    is_one_end_point_collinear = line_segment_2_start_orientation_relative_to_line_segment_1 * line_segment_2_end_orientation_relative_to_line_segment_1 * line_segment_1_start_orientation_relative_to_line_segment_2 * line_segment_1_end_orientation_relative_to_line_segment_2 == 0
+
+    could_line_segment_1_contain_line_segment_2_start = could_line_segment_contain_coordinate(line_segment_1,
+                                                                                              line_segment_2.start)
+    could_line_segment_1_contain_line_segment_2_end = could_line_segment_contain_coordinate(line_segment_1,
+                                                                                            line_segment_2.end)
+    could_line_segment_2_contain_line_segment_1_start = could_line_segment_contain_coordinate(line_segment_2,
+                                                                                              line_segment_1.start)
+    could_line_segment_2_contain_line_segment_1_end = could_line_segment_contain_coordinate(line_segment_2,
+                                                                                            line_segment_1.end)
+
+    return is_one_end_point_collinear and \
+           (could_line_segment_1_contain_line_segment_2_start or could_line_segment_1_contain_line_segment_2_end or
+            could_line_segment_2_contain_line_segment_1_start or could_line_segment_2_contain_line_segment_1_end)
+
+
+def calculate_point_triplet_orientation(coordinate_1, coordinate_2, coordinate_3):
+    """
+    Returns a value that represents if connecting coordinate_1 to coordinate_2 to coordinate_3 occurs in a
```

Comment:
This definition could be better I think this value really represents the `-sine` of the number of degrees that the three coordinates create